specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2018-02-01-preview/containerregistry_build.json

```diff
@@ -0,0 +1,2220 @@
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BuildUpdateParameters"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/registryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/Build"
+            }
+          },
+          "202": {
+            "description": "Success"
+          },
+          "404": {
+            "description": "Not Found"
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Builds_Update": {
+            "$ref": "./examples/BuildsUpdate.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/builds/{buildId}/getLogLink": {
+      "get": {
+        "tags": [
+          "Builds"
+        ],
+        "summary": "Get the links to the various build logs.",
+        "operationId": "Builds_GetLogLink",
+        "consumes": [
+          "application/json"
+        ],
+        "produces": [
+          "application/json"
+        ],
+        "parameters": [
+          {
+            "name": "buildId",
+            "in": "path",
+            "description": "The build id.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "name": "logParameters",
+            "in": "body",
+            "description": "The build log parameters.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/BuildLogParameters"
+            }
+          },
+          {
+            "$ref": "#/parameters/subscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/resourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/registryNameParameter"
+          },
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Success",
+            "schema": {
+              "$ref": "#/definitions/BuildLogResult"
+            }
+          },
+          "404": {
+            "description": "Not Found"
+          }
+        },
+        "x-ms-examples": {
+          "Builds_GetLogLink": {
+            "$ref": "./examples/BuildsGetLogLink.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.ContainerRegistry/registries/{registryName}/builds/queue": {
```

Comment:
IMO, this route is not quite RESTful ...

Reply:
This is a one such case where we don't have resource id or name when we create it like builds and triggers. Let us get some recommendations from the ARM team.

Reply:
The route will not work. You need a name segment after /builds

Reply:
This is one off case where this is an POST on builds that RP  will generate the build id and return as response. What is your recommendation?